app/src/main/java/com/beemdevelopment/aegis/ui/EditEntryActivity.java

```diff
@@ -627,6 +660,14 @@ private VaultEntry parseEntry() throws ParseException {
 
         String type = _dropdownType.getText().toString();
         String algo = _dropdownAlgo.getText().toString();
+        String lowerCasedType = type.toLowerCase(Locale.ROOT);
+
+        if (lowerCasedType.equals(YandexInfo.ID)) {
+            int pinLength = _textYandexPin.length();
+            if (pinLength < 4) {
+                throw new ParseException("Pin is a required field. Min 4 digits.");
```

Comment:
We should probably capitalize **PIN** any time we display it in the UI.

Reply:
Changed :)